File: 0x12-javascript-web_scraping/3-rectangle.js
#!/usr/bin/node

exports.Rectangle = function Rectangle (w, h) {
  if (isNaN(parseInt(w)) === false && isNaN(parseInt(h)) === false &&
      w > 0 && h > 0) {
    this.width = w;
    this.height = h;
  }
  this.print = function () {
    for (let i = 0; i < this.height; i++) {
      let x = '';
      for (let i = 0; i < this.width; i++) {
        x += 'X';
      }
      console.log(x);
    }
  };
};

File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3


class Rectangle:
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    def __str__(self):
        rectangle = ""
        for i in range(self.__height):
            rectangle += "{}\n".format("#" * self.__width)
        return rectangle

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, width):
        if isinstance(width, int) is False:
            raise TypeError("width must be an integer")
        if width < 0:
            raise TypeError("width must be >= 0")
        self.__width = width

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, height):
        if isinstance(height, int) is False:
            raise TypeError("height must be an integer")
        if height < 0:
            raise TypeError("height must be >= 0")
        self.__height = height

    def area(self):
        return self.__width * self.__height

    def perimeter(self):
        return (self.__width + self.__height) * 2
